pkg/services/ngalert/models/alert_rule.go

```diff
@@ -830,3 +830,11 @@ func (r *Record) Fingerprint() data.Fingerprint {
 	writeString(r.From)
 	return data.Fingerprint(h.Sum64())
 }
+
+// RuleStatus contains info about a rule's current evaluation state.
+type RuleStatus struct {
+	Health            string
+	LastError         error
+	EvaluatedAt       time.Time
+	EvaluatedDuration time.Duration
+}
```

Comment:
I chose a single method returning a struct of all relevant fields, rather than separate methods for each field like Prometheus (`.Health()`, `.LastError()`, etc). This helps to minimize state processing in the event that there are a lot of states. It's all in memory, but there can be quite a few of them, so it's nicer to calculate everything in one go.